### PR TITLE
[Merged by Bors] - chore: delay imports of Star

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

**Comment by semorrison** [comment]
> Have you considered going the other way? It seems like we could have a `Star.Defs` file could be very light-weight.

We should do that too! :-)

**Comment by semorrison** [comment]
> Also please keep the imports sorted

I'm assuming the rule is just alphabetic? I hadn't realised we were doing this.

**Comment by semorrison** [comment]
(The motivation here was being annoyed at encountering these imports when doing a minimization.)

**Review by mattrobball — APPROVED** [review]

File: Mathlib/Algebra/Star/Basic.lean [hunk 1]
```diff
@@ -3,15 +3,12 @@ Copyright (c) 2020 Scott Morrison. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Scott Morrison
 -/
-import Mathlib.Algebra.Field.Opposite
 import Mathlib.Algebra.Group.Invertible.Defs
 import Mathlib.Algebra.Regular.Basic
 import Mathlib.Algebra.Ring.Aut
 import Mathlib.Algebra.Ring.CompTypeclasses
-import Mathlib.Algebra.Field.Opposite
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Algebra/Star/Basic.lean:11]: Yes.

File: scripts/noshake.json [hunk 2]
```diff
@@ -323,8 +323,10 @@
   "Mathlib.Analysis.Distribution.SchwartzSpace": ["Mathlib.Tactic.MoveAdd"],
   "Mathlib.AlgebraicTopology.DoldKan.Notations":
   ["Mathlib.AlgebraicTopology.AlternatingFaceMapComplex"],
+  "Mathlib.Algebra.Star.Module": ["Mathlib.Algebra.Module.LinearMap.Star"],
```
[the remark this answers is not in the document]
Comment (semorrison) [reply scripts/noshake.json:326]: Yes.

File: Mathlib/Algebra/Star/Basic.lean [hunk 3]
```diff
@@ -465,12 +454,8 @@ abbrev starRingOfComm {R : Type*} [CommSemiring R] : StarRing R :=
 
 instance Nat.instStarRing : StarRing ℕ := starRingOfComm
```
[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Algebra/Star/Basic.lean:455]: Sorry, I'm not sure what you mean here. Could you clarify, thanks?

[the remark this answers is not in the document]
Comment (semorrison) [reply Mathlib/Algebra/Star/Basic.lean:455]: I'm happy to move these later if needed. I'm going to continue on to merging so my next PR doesn't get conflicted.

**Comment by semorrison** [comment]
bors merge

File: Mathlib/Algebra/Star/Basic.lean (hunk 3 above)
[the remark this answers is not in the document]
Comment (mattrobball) [reply Mathlib/Algebra/Star/Basic.lean:455]: Yes, my point was: if the spirit moves you, then these should be moved in a later PR

File: Mathlib/Algebra/Star/Rat.lean [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/-
+Copyright (c) 2024 Lean FRO. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison
```
Comment (eric-wieser) [inline Mathlib/Algebra/Star/Rat.lean:4]:
```suggestion
Copyright (c) 2024 Yael Dillies. All rights reserved.
Released under Apache 2.0 license as described in the file LICENSE.
Authors: Yael Dillies
```
https://github.com/leanprover-community/mathlib4/pull/10633

**Comment by eric-wieser** [comment]
bors r-

bors d+

File: Mathlib/Algebra/Module/LinearMap/Star.lean [hunk 5]
```diff
@@ -0,0 +1,21 @@
+/-
+Copyright (c) 2024 Lean FRO. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison
+-/
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/LinearMap/Star.lean:5]:
```suggestion
Copyright (c) 2021 Frédéric Dupuis. All rights reserved.
Released under Apache 2.0 license as described in the file LICENSE.
Authors: Frédéric Dupuis
-/
```
from https://github.com/leanprover-community/mathlib/pull/9875

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**